_posts/dusan_andric/2016-10-28-azure-openstack.md

```diff
@@ -0,0 +1,153 @@
+
+Once everything is installed you can start the storage emulator by searching for it from the start screen. It should open up a new cmd window. You can this verify that it is running:
+
+<figure>
+    <img src="{{site.urlimg}}dusan_andric/azure/azure1.png" />
+    <figcaption>Azure Storage Emulator up and running.</figcaption>
+</figure>
+
+### Networking
+
+***This step is important***. The Storage Emulator is configured to only accept connections from localhost by default. 
+This can be changed by editing the config file, but then the storage server will become unavailable localy from the Windows instance. 
+Also it appears that storage clients when run with emulation mode set to true will also attempt to connect to a storage service running locally. 
+
+So for the moment port forwarding will be our workaround. For this example my only concern will be the blob storage. 
+
+On the windows server we can use the `netsh` utility to create our port foward.
+
+~~~powershell
+netsh 1337 127.0.0.1 10000
+~~~
+
+You can verify the result of the command:
+
+<figure>
+    <img src="{{site.urlimg}}dusan_andric/azure/azure2.png" />
+    <figcaption>netsh output in powershell.</figcaption>
+</figure>
+
+On your unix machine that needs to connect as a client, you can do something like the following:
+
+~~~bash
+ssh -L 10000:windows-server:1337 me@localhost
+~~~
+
+I'll be the first to say that this is not ideal, however it does work.
+
+### Verify
+
+Now that the setup is done, let's actually do something with the emulator in the form of a simple test. 
+
+First thing I'm going to do is upload an image to the blob storage using the Azure Storage Explorer.
+The tool is cross platform and very handy. 
+[You can download it here.](http://storageexplorer.com)
+
+From the UI I uploaded an image:
+
+<figure>
+    <img src="{{site.urlimg}}dusan_andric/azure/azure3.png" />
+    <figcaption>Storage Explorer in action.</figcaption>
+</figure>
+
+From my local machine I then run this little python script to download the image.
+
+~~~python
+from azure.storage.blob import BlockBlobService
+
+block_blob_service = BlockBlobService(is_emulated=True,
+                                      account_name='devstoreaccount1',
+                                      account_key='Eby8vdM02xNOcqFlqUwJPLlmEtlCDXJ1OUzFT50uSRZ6IFsuFq2UVErCz4I6tq/K1SZFPTOtr/KBHBeksoGMGw==')
+
+img_blob = block_blob_service.get_blob_to_bytes(container_name='data', blob_name='OICR2logo.png')
+open('OICR2logo.png', mode='wb').write(img_blob.content)
+~~~
+
+And there it is:
+
+<figure>
+    <img src="{{site.urlimg}}dusan_andric/azure/download.png" />
+    <figcaption>Successful download of image.</figcaption>
+</figure>
+
+## Conclusion
+
+I think it is safe to say that Microsoft products and services are no longer scary when coming from a Unix environment.
```

Comment:
Well, a fair bit of this article explains a "workaround" to connect. I would call it no longer scary once Microsoft implement a "no-workaround" solution :p

Note: This comment is a troll ...